typescript/lib/services/container_creation_config.ts

```diff
@@ -0,0 +1,119 @@
+const defaultKurtosisVolumeMountpoint = "/kurtosis-enclave-data";
+
+export type StaticFileID = string;
+
+// The ID of an artifact containing files that should be mounted into a service container
+export type FilesArtifactID = string;
+
+// ====================================================================================================
+//                                    Config Object
+// ====================================================================================================
+// TODO defensive copy when we're giving back complex objects?????
+// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+class ContainerCreationConfig {
+	
+    private readonly image: string;
+	private readonly kurtosisVolumeMountpoint: string;   // Technically the enclave data volume, but we call it this for simplicity for the user
+	private readonly usedPortsSet: Map<string, boolean>
+	private readonly fileGeneratingFuncs: Map<string, (func: number) => Error>; // TODO (comment) = file descriptors are just integers
```

Comment:
1)
oooo def leave that "file descriptors in Node are just integers" comment in place
2) is the `func` naming correct? should that be `fp` instead?